### ISSUE TITLE
Pub/Sub doesn't take namespace into account

### ISSUE THREAD

**Issue by torwig** [apache/kvrocks#952]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-kvrocks/issues) and found no similar issues.

### Version

Latest from the `unstable` branch, Fedora 35.

### Minimal reproduce step

Run `kvrocks` with additional two namespaces, e.g.

```
namespace.one password4one
namespace.two password4two
```

Run two `redisc-cli` instances, and connect to `kvrocks`. 
Switch to the first namespace via the first `redis-cli`: `auth password4one`.
Switch to the second namespace via the second `redis-cli`: `auth password4two`.
Execute `subscribe channel` via the second `redisc-cli`.
Execute `publish channel not-for-second-ns` via the first `redis-cli`.

### What did you expect to see?

The message, published in `namespace.one` should not be received by the subscribers in `namespace.two`, even if the channel names are the same.

### What did you see instead?

Subscribers from `namespace.two` received the message, published in `namespace.one`.
For Pub/Sub namespaces aren't isolated.

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by tanruixiang** [comment]
Good Catch. It seems that the reason for the problem is that the namespace information in `Connection` is not used.

**Comment by git-hulk** [comment]
I'm also wondering if any users need this feature

**Comment by manchurio** [comment]
I think this is ok, so does redis.

**Comment by git-hulk** [comment]
> I think this is ok, so does redis.

Yes, this behavior is matched the Redis if we regard the namespace as the Redis DB. So I'm wondering if any users need this feature.

**Comment by tanruixiang** [comment]
> Yes, this behavior is matched the Redis if we regard the namespace as the Redis DB. So I'm wondering if any users need this feature.

If someone needs this feature, I think they can just use stream.

**Comment by git-hulk** [comment]
> > Yes, this behavior is matched the Redis if we regard the namespace as the Redis DB. So I'm wondering if any users need this feature.
> 
> If someone needs this feature, I think they can just use stream.

There's a bit different between stream and pubsub. For the stream, it likes most message queue system, all message can be saw before the retention ttl but the pubsub is fire and forget.

**Comment by tanruixiang** [comment]
> There's a bit different between stream and pubsub. For the stream, it likes most message queue system, all message can be saw before the retention ttl but the pubsub is fire and forget.

Yes, I mean that if someone needs this feature, they can use stream instead in the current situation. In fact, many people who directly use Redis also use stream instead of pubsub to avoid the disadvantages of pubsub. I think it is enough for kvrocks to align with Redis.
